dotnet-testing/MockingWithNSubstitute/MockingWithNSubstitute.Tests/NotificationServiceTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿using FluentAssertions;
+using NSubstitute;
+
+namespace MockingWithNSubstitute.Tests
+{
+    public class NotificationServiceTests
+    {
+        private readonly User _user;
+        private readonly IEmailService _emailService;
+        private readonly NotificationService _notificationService;
+
+        public NotificationServiceTests()
+        {
+            _user = new User("Code-Maze", "email@code-maze.com");
+            _emailService = Substitute.For<IEmailService>();
+            _notificationService = new NotificationService(_emailService);
+        }
+
+        [Fact]
+        public void GivenInputIsCorrect_WhenNotifyUserIsInvoked_ThenTrueIsReturned()
+        {
+            // Arrange            
+            var message = "Mocking behaviors and expectations with NSubstitute";
+            _emailService.IsValidEmail(_user.Email)
+                .Returns(true);
+            _emailService.SendEmail(_user.Email, "Notification from CodeMaze", message)
+                .Returns(true);
+
+            // Act
+            var result = _notificationService.NotifyUser(_user, message);
+
+            // Assert
+            result.Should().BeTrue();
+        }
+
+        [Fact]
+        public void GivenInputIsNotCorrect_WhenNotifyUserIsInvoked_ThenFalseIsReturned()
+        {
+            // Arrange
+            var message = "Ignoring or Conditionally Matching Arguments";
```

Comment:
This can be made const `const string message =`

Reply:
Sorted